doc extlink URL fixes

hflynn opened the pull request:
Updates the Model docs extlink to use the latest redirect now that works and also I spotted that I'd updated the BF doc link in the contributing docs conf.py but not the OMERO one so I've moved it to common conf.py (since both doc sets use it) and updated it.

Commit:
doc extlink URL fixes

sbesson commented on common/conf.py:
```diff
@@ -155,10 +155,10 @@ def get_previous_version(majornumber, minornumber=0):
+    # Doc links
+    'model_doc' : (docs_root + '/latest/ome-model/%s', ''),
     'devs_doc' : (docs_root + '/contributing/%s', ''),
+    'bf_doc' : (docs_root + '/latest/bio-formats/%s', ''),
```
Is this used in the contributing documentation only or in the OMERO documentation? In the former, this makes sense as the contributing documentation largely documents our latest software. For OMERO, arguably, we should be able to point at the documentation for the specific version of Bio-Formats shipped with OMERO.

hflynn replied:
It's in both but I can make the OMERO one version specific if we prefer.

hflynn replied:
That would mean we'd need to update it regularly though.

sbesson replied:
Agreed. The main reason for hardcoding from my side is that https://github.com/openmicroscopy/ome-documentation/blob/v5.4.0-m3/omero/users/cli/import.rst refers to supported formats table which is tightly coupled to the the version included into OMERO.

Some of the autogeneration scripts read properties from the OMERO version (patch versions) for consumption in the docs, should we consider adding the bf_version there as well to make sure it's updated via the auto-generation PR?

sbesson committed:
Also read versions.bioformats from etc/omero.properties

sbesson committed:
Update conf_autogen.py

Generated using ./omero/autogen_db_version.py ~/Downloads/OMERO.server-5.4.0-m3-401-09e22ed-ice36-b756/ > omero/conf_autogen.py

Commit:
Merge pull request #28 from sbesson/bf_version

Read Bio-Formats version

Commit:
use versioned BF docs for OMERO only

hflynn commented:
BF doc extlinks moved back into their respective subfolders so we can use latest for the contributing docs and an autogen'ed BF version for OMERO.

Likely we'll want to have similar for the Model docs in future.

Commit:
remove /site from URL in README

jburel commented on README.rst:
```diff
@@ -436,7 +436,7 @@ ZeroC                       https://zeroc.com
 Jenkins                     http://jenkins-ci.org
 roadmap                     https://trac.openmicroscopy.org/ome/roadmap
-Open Microscopy Environment http://www.openmicroscopy.org/site
+Open Microscopy Environment http://www.openmicroscopy.org
```
https?

jburel replied:
same for gs

jburel commented on omero/conf_autogen.py:
```diff
@@ -1,4 +1,5 @@
 omero_db_version = "OMERO5.4DEV"
 omero_db_patch = "3"
+versions_bioformats = "5.6.0"
```
Could that value be extracted from ``omero.properties``?

hflynn replied:
I thought that was what Seb had done

jburel replied:
done in ``autogen_db_version.py``

hflynn replied:
@sbesson I'll leave this one to you

sbesson replied:
Isn't that done in c132c4f0cc2244df1019a7b87f104e45a0d447f6? See  the commit message of 741450ce83dd4bef349ecb5fc7981529a948ef5d

jburel replied:
@hflynn sorry that's what I meant by done in ``autogen_db_version.py``
I first missed it
no action required

jburel commented:
2 remaining ``:bf_doc:`` in ``contributing/data-model-schema.rst and contributing/index.rst`` that will point Do you want to keep them?

Commit:
make website URLs https

hflynn commented:
@jburel the remaining bf_docs links in the contributing docs are fine - they are set to latest in the conf.py for contributing as those docs are consistently documenting the latest state of play. I've changed the OMERO ones specifically so that the supported formats table is always showing the right info for the current version.

jburel commented:
Great! I was not sure if they were missed or not
